pkg/query/intersection_arrow_test.go

```diff
@@ -0,0 +1,503 @@
+package query
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/authzed/spicedb/internal/datastore/memdb"
+	"github.com/authzed/spicedb/pkg/datastore"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+func TestIntersectionArrowIterator(t *testing.T) {
+	t.Parallel()
+
+	require := require.New(t)
+
+	t.Run("AllSubjectsSatisfyCondition", func(t *testing.T) {
+		t.Parallel()
+
+		// Left side: document has teams team1 and team2
+		leftPath1 := MustPathFromString("document:doc1#team@team:team1")
+		leftPath2 := MustPathFromString("document:doc1#team@team:team2")
+
+		// Right side: alice is member of both team1 and team2
+		rightPath1 := MustPathFromString("team:team1#member@user:alice")
+		rightPath2 := MustPathFromString("team:team2#member@user:alice")
+
+		leftIter := NewFixedIterator(leftPath1, leftPath2)
+		rightIter := NewFixedIterator(rightPath1, rightPath2)
+
+		intersectionArrow := NewIntersectionArrow(leftIter, rightIter)
+
+		// Create test context
+		ds, err := memdb.NewMemdbDatastore(0, 0, memdb.DisableGC)
+		require.NoError(err)
+
+		revision, err := ds.ReadWriteTx(context.Background(), func(ctx context.Context, tx datastore.ReadWriteTransaction) error {
+			return nil
+		})
+		require.NoError(err)
+
+		ctx := &Context{
+			Context:   context.Background(),
+			Executor:  LocalExecutor{},
+			Datastore: ds,
+			Revision:  revision,
+		}
+
+		// Test: alice should have access because she's a member of ALL teams (team1 and team2)
+		resources := []Object{NewObject("document", "doc1")}
+		subject := ObjectAndRelation{ObjectType: "user", ObjectID: "alice"}
+
+		pathSeq, err := ctx.Check(intersectionArrow, resources, subject)
+		require.NoError(err)
+
+		paths, err := CollectAll(pathSeq)
+		require.NoError(err)
+
+		// Should return a single result since alice is in ALL teams (intersection semantics)
+		require.Len(paths, 1, "Should return single path representing the intersection")
+
+		// Verify the result has alice as the subject
+		require.Equal("alice", paths[0].Subject.ObjectID, "Result should have alice as the subject")
```

Comment:
if you print `paths[0].Relation` you get `team`, which seems arbitrary. should this not be an empty string?